iroh-resolver/src/resolver.rs

```diff
@@ -1678,6 +1700,129 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_resolver_seeking() {
+        // Test content
+        // ------------
+        // QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN foo/hello.txt
+        //   contains: "hello"
+        // QmdkGfDx42RNdAZFALHn5hjHqUq7L9o6Ef4zLnFEu3Y4Go foo
+
+        let hello_txt_cid_str = "QmZULkCELmmk5XNfCgTnCyFgAVxBRBXyDHGGMVoLFLiXEN";
+        let hello_txt_block_bytes = load_fixture(hello_txt_cid_str).await;
+
+        // read root
+        let root_cid_str = "QmdkGfDx42RNdAZFALHn5hjHqUq7L9o6Ef4zLnFEu3Y4Go";
+        let root_cid: Cid = root_cid_str.parse().unwrap();
+        let root_block_bytes = load_fixture(root_cid_str).await;
+
+        let loader: HashMap<Cid, Bytes> = [
+            (root_cid, root_block_bytes.clone()),
+            (hello_txt_cid_str.parse().unwrap(), hello_txt_block_bytes),
+        ]
+        .into_iter()
+        .collect();
+        let loader = Arc::new(loader);
+        let resolver = Resolver::new(loader.clone());
+
+        let path = format!("/ipfs/{root_cid_str}/hello.txt");
+        let ipld_hello_txt = resolver.resolve(path.parse().unwrap()).await.unwrap();
+
+        if let OutContent::Unixfs(node) = ipld_hello_txt.content {
+            // clip response
+            let cr = seek_and_clip(&node, resolver.clone(), 0..2).await;
+            assert_eq!(read_to_string(cr).await, "he");
+
+            let cr = seek_and_clip(&node, resolver.clone(), 0..5).await;
+            assert_eq!(read_to_string(cr).await, "hello");
+
+            // clip to the end
+            let cr = seek_and_clip(&node, resolver.clone(), 0..6).await;
+            assert_eq!(read_to_string(cr).await, "hello\n");
+
+            // clip beyond the end
+            let cr = seek_and_clip(&node, resolver.clone(), 0..100).await;
+            assert_eq!(read_to_string(cr).await, "hello\n");
+
+            // seek
+            let cr = seek_and_clip(&node, resolver.clone(), 1..100).await;
+            assert_eq!(read_to_string(cr).await, "ello\n");
+
+            // seek and clip
+            let cr = seek_and_clip(&node, resolver.clone(), 1..3).await;
+            assert_eq!(read_to_string(cr).await, "el");
+        } else {
+            panic!("invalid result: {:?}", ipld_hello_txt);
+        }
+    }
+
+    #[tokio::test]
```

Comment:
Either in this PR or at least have an issue for seeking across dag-cbor, dag-json and raw files as well please.

Reply:
https://github.com/n0-computer/iroh/issues/346

Reply:
Gotta do other stuff before release :/